test/integration/test_push_docker_image.py

```diff
@@ -24,32 +24,50 @@ def normalize_request_name(name: str):
     return re.sub(r"^_+|_+$", "", name)
 
 
+@dataclass
+class DockerImageSpec:
+    repository: str
+    tag: str
+
+    @property
+    def name(self) -> str:
+        return f"{self.repository}:{self.tag}"
+
+
+@pytest.fixture(scope="session")
+def registry_image():
+    spec = DockerImageSpec("registry", "2")
+    client = docker.from_env()
+    if not client.images.list(spec.name):
+        _logger.debug("Pulling Docker image with Docker registry")
+        client.images.pull(spec.repository, spec.tag)
+    return spec
+
+
+@pytest.fixture(scope="session")
+def sample_docker_image(registry_image):
+    return registry_image
+
+
 @contextlib.contextmanager
-def tagged_image(
-        dss_docker_image: DssDockerImage,
-        registry: LocalDockerRegistry,
-        tag: str = None,
-):
+def tagged_image(old: DockerImageSpec, new: DockerImageSpec):
     """
     Prepend host and port of LocalDockerRegistry to the repository name of the
```

Comment:
this description is outdated

Reply:
Ouch - you're right. Will be fixed in next push.